lesson_7/mihaela_vaida/print_register_dec.py

```diff
@@ -0,0 +1,30 @@
+print_registry = []
+
+
+def register(fnc):
+    def print_register(arg):
```

Comment:
Is the inner function really needed?
Justify the answer, whichever that might be

Reply:
I think it is , because  otherwise, a traceback error is thrown!

Reply:
Try harder ;)

Reply:
Is it possible to have a decorator without an inner function ? I google that and the examples of decorators have one, but I am not sure about the answer.

Reply:
Read extra carefully the provided definition

Reply:
But the decorated functions have arguments, that`s why I think the inner function is needed for the decorator

Reply:
print_registry = []

def register(fnc):

    return print(print_registry.append(fnc.__name__))

@register
def greet(name):
    return "Greetings {}!".format(name)


def say_hello(name):
    return "Hello {}!".format(name)

@register
def say_goodbye(name):
    return "Goodbye {}!".format(name)



print(say_hello('Vio'))  # 'Hello Vio!'

print(print_registry) #['greet', 'say_goodbye']

Reply:
I tried this update and it works like that, but   if  I call  say_goodbye('name')/ greet('dsd') , then an error is thrown

Reply:
please let me know if it is ok  and I'll do the commit and push :)